From 46d4dc2b52931f77dc8ccd82d76b56a7bc2a6fc1 Mon Sep 17 00:00:00 2001
Message-Id: <46d4dc2b52931f77dc8ccd82d76b56a7bc2a6fc1.1435870144.git.chang-joon.lee@intel.com>
From: Gajanan Bhat <gajanan.bhat@intel.com>
Date: Tue, 30 Jun 2015 16:16:03 +0530
Subject: [PATCH] SQUASHME! [VPG]: Fix KW issue - check if CRTC is NULL

This patch is to fix Klockwork issue which is complaining that
CRTC can be NULL. Though this condition is not likely to happen,
this patch addresses these warnings.

SQUASHME! - This patch should be squashed to the following patch:

	commit 5f3f638dda315e1678e17d99a4446e27cf382027
	Change-Id: I45fba036a4e5aec8805cf2e4dfa26a4665b9e548
	Author: Gajanan Bhat <gajanan.bhat@intel.com>
	Date:   Tue May 26 15:40:42 2015 +0200
		MUST_REBASE [VPG]: drm/i915: Add connector property for forcing
		low ddr frequency

Issue: GMINL-12381
Change-Id: Id2f456c17f7f10492a8eb22f52cce50a7ef227a1
Signed-off-by: Gajanan Bhat <gajanan.bhat@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 61a5793..f9021aa 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -1640,11 +1640,11 @@ void vlv_set_ddr_dvfs(struct drm_i915_private *dev_priv,
 	if (dev_priv->force_low_ddr_freq)
 		val = CHV_FORCE_DDR_LOW_FREQ;
 
+	crtc = single_enabled_crtc(dev_priv->dev);
 	/* DDR freq should be high if more than one pipe is active */
-	if (!single_pipe_enabled(pipe_stat))
+	if (!single_pipe_enabled(pipe_stat) || crtc == NULL)
 		val = CHV_FORCE_DDR_HIGH_FREQ;
 	else {
-		crtc = single_enabled_crtc(dev_priv->dev);
 		config = &((to_intel_crtc(crtc))->config);
 
 		/* DDR freq should be high for resolution greater than 19x12 */
-- 
1.9.1

